### ISSUE TITLE
Feature Request MultiQuantile

### ISSUE THREAD

**Issue by 1991jhf** [Evovest/EvoTrees.jl#269]
It would be nice to have multi-quantile regression for approximating histogram in one go.
similar to
https://catboost.ai/en/docs/concepts/loss-functions-regression#MultiQuantile

**Comment by jeremiedb** [comment]
Thanks for the pointer. 

The part I found tricky in supporting quantiles is that as the loss isn't differentiable per se, I ended up tracking full vector in observations/targets in order to compute the leaf prediction values. Having a reliable gradient based proxy would be preferable. Notably for maintenance as I for now don't have much use case with such loss objectives. 
If you can point to a gradient based approach for estimating the loss/gain and associated leaf predictions, it should be fairly strightforward to add support for this loss.

**Comment by benjamin-jsauve** [comment]
Hi Jérémie, looked into #269 and I think the path forward is a sigmoid-smoothed pinball loss.

Replace the indicator $\mathbf{1}[y < \hat{y}_k]$ with $\sigma((\hat{y}_k - y)/\delta)$, giving a proper Hessian

$$h_{i,k} = \frac{1}{\delta} \sigma\left(\frac{\hat{y}_{i,k} - y_i}{\delta}\right)\left(1 - \sigma\left(\frac{\hat{y}_{i,k} - y_i}{\delta}\right)\right)$$

Leaf values then fall out of the standard $w^* = -G_k/H_k$ without any within-leaf quantile tracking (this is basically what CatBoost does). So then

1. Should $\delta$ be user-facing or auto-scaled to $\text{std}(y)$?
2. For quantile crossing, do you think simple isotonic post-processing is enough for now, or is it worth enforcing constraints directly at the leaf level?

**Comment by jeremiedb** [comment]
@benjamin-jsauve Is such sigmoid-smoothed pinball loss actually used in CatBoost? 
I thought CatBoost and the likes simplused used vanilla pinball loss. A caveat with pinball loss, and I suspect any variations of it like a smoothed one, is the missmatched calibration between the learned predictor and the actual quantile. Ie: the a 80th-percentile is often quite off being >= 80% of observed target. 

I'm not having much bandwidth to work on this. But if there's evidence of added value compared to generic pinball and that it behaves well; feed free to open a PR.

**Comment by benjamin-jsauve** [comment]
You're correct, I misspoke. I meant to cite the arctan pinball loss (arXiv:2406.02293). I think keeping the exact residual approach but isolating the sorting to only the terminal leaf step, completly decoupled from the histogram pipeline is the way to go as this makes the maintenance burden go away without sacrificing calibration which any smooth proxy would. Although it could make for an interesting benchmark. Happy to open a PR along those lines if that works for you.
